proxy/healthy_endpoints_test.go

```diff
@@ -188,3 +181,41 @@ func TestPHCForMultipleHealthyAndOneUnhealthyEndpoints(t *testing.T) {
 		assert.InDelta(t, 0, failedReqs, 0.1*float64(nRequests))
 	})
 }
+
+func TestPHCForMultipleHealthyAndMultipleUnhealthyEndpoints(t *testing.T) {
+	services := setupServices(t, 2, 2)
+	for _, algorithm := range []string{"random", "consistentHash", "roundRobin", "powerOfRandomNChoices"} {
+		t.Run(algorithm, func(t *testing.T) {
+
+			endpointRegistry := routing.NewEndpointRegistry(routing.RegistryOptions{
+				PassiveHealthCheckEnabled:     true,
+				StatsResetPeriod:              period,
+				MinRequests:                   1, // with 3 test case fails
+				MaxHealthCheckDropProbability: 1.0,
+				MinHealthCheckDropProbability: 0.01,
+			})
+
+			mockMetrics, ps := setupProxyWithCustomEndpointRegisty(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s", "%s">`,
+				algorithm, services[0].URL, services[1].URL, services[2].URL, services[3].URL), endpointRegistry)
+			failedReqs := sendGetRequests(t, ps)
+			assert.InDelta(t, 0, failedReqs, 0.1*float64(nRequests))
+			mockMetrics.WithCounters(func(counters map[string]int64) {
+				assert.InDelta(t, float64(nRequests)*2.0, float64(counters["passive-health-check.endpoints.dropped"]), 0.3*float64(nRequests)) // allow 30% error
+			})
+		})
+	}
+
+	t.Run("consistent hash with balance factor", func(t *testing.T) {
+		endpointRegistry := routing.NewEndpointRegistry(routing.RegistryOptions{
+			PassiveHealthCheckEnabled:     true,
+			StatsResetPeriod:              period,
+			MinRequests:                   1, // with 3 test case fails
```

Comment:
I guess `sendGetRequests()` is too slow or the backend is too slow.
Maybe use vegeta as library to create a stable amount of requests like 10rps.

Reply:
So it does not do concurrent requests and it reads the response and checks for error and status code.
In a CI environment that's likely too slow for having MinRequests increased to 3. Vegeta would be the better option in this case.

Reply:
yes that was the problem and we were looking how to make send them concurrently Vegeta seems like a good option